compiler/quilt/test/test_import.py

```diff
@@ -198,6 +198,130 @@ def test_multiple_package_dirs(self):
             from quilt.data.foo import multiple1
             assert multiple1.dataframes
 
+             # Check search order again just to make sure it doesn't return same named empty package
+            from quilt.data.foo import multiple1
+            assert multiple1.dataframes
+
+            teststore = PackageStore(self._store_dir)
+            contents = open(os.path.join(teststore.package_path(None, 'foo', 'multiple1'),
+                                          Package.CONTENTS_DIR,
+                                          multiple1._package.get_hash())).read()
+            assert contents
+
+            # bad imports
+            with self.assertRaises(ImportError):
+                from quilt.data.nonexisting import multiple2
+            with self.assertRaises(ImportError):
+                from quilt.data.foo import nonexistingdata
+            with self.assertRaises(ImportError):
+                import quilt.data.foo.nonexisting
+            with self.assertRaises(ImportError):
+                import quilt.data.bad_user.nonexisting
+            with self.assertRaises(ImportError):
+                from quilt.data.foo.dataframes import blah
+            with self.assertRaises(ImportError):
+                from quilt.data.foo.baz import blah
+
+        # add same dir to the search path
+        dirs = 'foo/%s:%s:%s' % (PACKAGE_DIR_NAME, new_build_dir, new_build_dir)
+        with patch.dict(os.environ, {'QUILT_PACKAGE_DIRS': dirs}):
+            from quilt.data.foo import multiple2
```

Comment:
Doesn't do anything because the import is cached

Reply:
@Kurlov pls remove and add a comment ("no value in testing same dir in search path because of python import caching")

Reply:
@asah: Not quite; my point was, regardless of what `QUILT_PACKAGE_DIRS` might be or any other setting, this particular import would succeed (even when it shouldn't) because of caching.

That said, testing duplicates in `QUILT_PACKAGE_DIRS` is probably not that useful - I can't really think of a bug that might cause that to fail.